modules/plugins/languages/scala.nix

```diff
@@ -0,0 +1,138 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}: let
+  inherit (lib) types optionalString mkPackageOption;
```

Comment:
We don't do `inherit (lib) types`. Please inherit each type that you use

`optionalString` is from `lib.strings`, `mkPackageOption` is from `lib.options` which is already in the scope - can be moved after `mkEnableOption`.